app/controllers/admin/email_domain_blocks_controller.rb

```diff
@@ -6,12 +6,12 @@ class EmailDomainBlocksController < BaseController
 
     def index
       authorize :email_domain_block, :index?
-      @email_domain_blocks = EmailDomainBlock.page(params[:page])
+      @email_domain_blocks = EmailDomainBlock.where(parent_id: nil).includes(:children).order(id: :desc).page(params[:page])
     end
 
     def new
       authorize :email_domain_block, :create?
-      @email_domain_block = EmailDomainBlock.new
+      @email_domain_block = EmailDomainBlock.new(domain: params[:_domain])
```

Comment:
why `:_domain` vs `:domain`?

Reply:
That's the query string param. It can't be `domain` because that's a reserved keyword in `link_to`

Reply:
ok